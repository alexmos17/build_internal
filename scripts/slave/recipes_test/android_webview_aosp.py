# Copyright 2013 The Chromium Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

"""Test inputs for recipes/android_webview_aosp.py"""


def _common_factory_properties(**kwargs):
  ret = {
    'android_repo_url': 'https://android.googlesource.com/platform/manifest',
    'android_repo_branch': 'master',
  }
  ret.update(kwargs)
  return ret

def _common_test_data():
  return {
    'calculate trimmed deps': (0, {
        'blacklist': {
          'src/blacklist/project/1': None,
          'src/blacklist/project/2': None,
      }
    })
  }

def basic_test(api):
  return  {
    'factory_properties': _common_factory_properties(),
    'test_data': _common_test_data(),
  }

def with_ndk_pin_revision_test(api):
  return  {
    'factory_properties': _common_factory_properties(
      android_ndk_pin_revision='5049b437591600fb0d262e4215cee4226e63c6ce'
    ),
    'test_data': _common_test_data(),
  }

def with_resync_projects_test(api):
  return  {
    'factory_properties': _common_factory_properties(
      android_repo_resync_projects=['frameworks/base']
    ),
    'test_data': _common_test_data(),
  }

def uses_android_repo_test(api):
  return {
    'factory_properties': _common_factory_properties(),
    'test_data': _common_test_data(),
    'paths_to_mock' : [
        '[SLAVE_BUILD_ROOT]/android-src/.repo/repo/repo',
        '[SLAVE_BUILD_ROOT]/android-src',
    ],
  }

def does_delete_stale_chromium_test(api):
  return {
    'factory_properties': _common_factory_properties(),
    'test_data': _common_test_data(),
    'paths_to_mock' : [
        '[SLAVE_BUILD_ROOT]/android-src/external/chromium_org',
    ],
  }

def uses_goma_test(api):
  return {
    'factory_properties': _common_factory_properties(),
    'test_data': _common_test_data(),
    'paths_to_mock' : ['[BUILD_ROOT]/goma', ]
  }

def uses_svn_revision_if_present_test(api):
  return  {
    'build_properties': { 'revision': 1234 },
    'factory_properties': _common_factory_properties(),
    'test_data': _common_test_data(),
  }
