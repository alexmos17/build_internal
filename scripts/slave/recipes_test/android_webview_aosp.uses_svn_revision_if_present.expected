[
  {
   "cmd": ["[DEPOT_TOOLS_ROOT]/gclient", "config", "--spec", "deps_os = {}\nhooks = []\ntarget_os_only = False\nsolutions = [{'managed': True, 'custom_vars': {}, 'url': 'svn://svn-mirror.golo.chromium.org/chrome/trunk/src', 'custom_deps': {}, 'deps_file': 'DEPS', 'safesync_url': None, 'name': 'src'}]\ntarget_os = []\n"],
   "name": "[spec: bare] gclient setup",
   "seed_steps": ["[spec: bare] gclient setup", "[spec: bare] gclient clean", "[spec: bare] gclient sync"]
  },
  {
   "cmd": ["[DEPOT_TOOLS_ROOT]/gclient", "revert", "--nohooks"],
   "name": "[spec: bare] gclient clean"
  },
  {
   "cmd": ["[DEPOT_TOOLS_ROOT]/gclient", "sync", "--nohooks", "--revision", "src@1234"],
   "name": "[spec: bare] gclient sync"
  },
  {
   "cmd": ["[SLAVE_BUILD_ROOT]/src/android_webview/buildbot/deps_whitelist.py", "--method", "android_build", "--path-to-deps", "[SLAVE_BUILD_ROOT]/src/DEPS", "--output-json", "/path/to/tmp/json"],
   "name": "calculate trimmed deps"
  },
  {
   "cmd": ["[DEPOT_TOOLS_ROOT]/gclient", "config", "--spec", "deps_os = {}\nhooks = []\ntarget_os_only = False\nsolutions = [{'managed': True, 'custom_vars': {}, 'url': 'svn://svn-mirror.golo.chromium.org/chrome/trunk/src', 'custom_deps': {'src/blacklist/project/2': None, 'src/blacklist/project/1': None}, 'deps_file': 'DEPS', 'safesync_url': None, 'name': 'src'}]\ntarget_os = ['android']\n"],
   "name": "[spec: trimmed] gclient setup",
   "seed_steps": ["[spec: trimmed] gclient setup", "[spec: trimmed] gclient clean", "[spec: trimmed] gclient sync"]
  },
  {
   "cmd": ["[DEPOT_TOOLS_ROOT]/gclient", "revert", "--nohooks"],
   "name": "[spec: trimmed] gclient clean"
  },
  {
   "cmd": ["[DEPOT_TOOLS_ROOT]/gclient", "sync", "--nohooks", "--revision", "src@1234"],
   "name": "[spec: trimmed] gclient sync"
  },
  {
   "cmd": ["[SLAVE_BUILD_ROOT]/src/build/util/lastchange.py", "-o", "[SLAVE_BUILD_ROOT]/src/build/util/LASTCHANGE", "-s", "[SLAVE_BUILD_ROOT]/src"],
   "name": "Chromium LASTCHANGE",
   "seed_steps": ["Chromium LASTCHANGE", "Blink LASTCHANGE"]
  },
  {
   "cmd": ["[SLAVE_BUILD_ROOT]/src/build/util/lastchange.py", "-o", "[SLAVE_BUILD_ROOT]/src/build/util/LASTCHANGE.blink", "-s", "[SLAVE_BUILD_ROOT]/src/third_party/WebKit"],
   "name": "Blink LASTCHANGE"
  },
  {
   "cmd": ["mkdir", "[SLAVE_BUILD_ROOT]/android-src"],
   "name": "mkdir android source root",
   "seed_steps": ["mkdir android source root", "repo init"]
  },
  {
   "cmd": ["[DEPOT_TOOLS_ROOT]/repo", "init", "-u", "https://android.googlesource.com/platform/manifest", "-b", "master"],
   "cwd": "[SLAVE_BUILD_ROOT]/android-src",
   "name": "repo init"
  },
  {
   "cmd": ["[SLAVE_BUILD_ROOT]/src/android_webview/buildbot/generate_local_manifest.py", "[SLAVE_BUILD_ROOT]/android-src", "external/chromium_org"],
   "name": "generate local manifest"
  },
  {
   "cmd": ["[DEPOT_TOOLS_ROOT]/repo", "sync", "-j16", "-d", "-f"],
   "cwd": "[SLAVE_BUILD_ROOT]/android-src",
   "name": "repo sync",
   "seed_steps": ["repo sync"]
  },
  {
   "cmd": ["ln", "-s", "[SLAVE_BUILD_ROOT]/src", "[SLAVE_BUILD_ROOT]/android-src/external/chromium_org"],
   "name": "symlink chromium_org",
   "seed_steps": ["symlink chromium_org"]
  },
  {
   "cmd": ["[BUILD_ROOT]/scripts/slave/android/with_lunch", "[SLAVE_BUILD_ROOT]/android-src", "full-eng", "[SLAVE_BUILD_ROOT]/android-src/external/chromium_org/android_webview/tools/gyp_webview"],
   "cwd": "[SLAVE_BUILD_ROOT]/android-src/external/chromium_org",
   "name": "gyp_webview",
   "seed_steps": ["gyp_webview"]
  },
  {
   "cmd": ["[BUILD_ROOT]/scripts/slave/android/with_lunch", "[SLAVE_BUILD_ROOT]/android-src", "full-eng", "[BUILD_ROOT]/scripts/slave/compile.py", "libwebviewchromium", "android_webview_java", "--build-dir", "[SLAVE_BUILD_ROOT]", "--src-dir", "[SLAVE_BUILD_ROOT]/android-src", "--target-output-dir", "[SLAVE_BUILD_ROOT]/android-src/out", "--build-tool", "make-android", "--verbose"],
   "cwd": "[SLAVE_BUILD_ROOT]",
   "name": "compile",
   "seed_steps": ["compile"]
  }
]